video/out/opengl/hwdec_cuda.c

```diff
@@ -197,6 +263,84 @@ static void cuda_uninit(struct ra_hwdec *hw)
 #undef CHECK_CU
 #define CHECK_CU(x) check_cu((mapper)->owner, (x), #x)
 
+static struct ra_buf *cuda_buf_pool_get(struct ra_hwdec_mapper *mapper, int n)
+{
+    struct priv_owner *p_owner = mapper->owner->priv;
+    struct priv *p = mapper->priv;
+    CudaFunctions *cu = p_owner->cu;
+    int ret = 0;
+
+    struct ra_buf_pool *pool = &p->buf_pool[n];
+    struct ra_buf *buf = ra_buf_pool_get(mapper->ra, pool, &p->buf_params[n]);
+    if (!buf) {
+        goto error;
+    }
+
+    if (!ra_vk_buf_get_priv(buf)) {
+        struct ext_buf *ebuf = talloc_zero(NULL, struct ext_buf);
+
+        struct vk_external_fd fd_info;
+        bool success = ra_vk_buf_get_external_fd(mapper->ra, buf, &fd_info);
+        if (!success) {
+            ret = -1;
+            goto error;
+        }
+
+        ebuf->fd = fd_info.mem_fd;
+        MP_DBG(mapper, "vk_external_fd[%d][%d]: %d %lu %lu\n", n, pool->index, ebuf->fd, fd_info.size, fd_info.offset);
+
+        CUDA_EXTERNAL_MEMORY_HANDLE_DESC ext_desc = {
+            .type = CU_EXTERNAL_MEMORY_HANDLE_TYPE_OPAQUE_FD,
+            .handle.fd = ebuf->fd,
+            .size = fd_info.mem_size,
+            .flags = 0,
+        };
+        ret = CHECK_CU(cu->cuImportExternalMemory(&ebuf->mem, &ext_desc));
+        if (ret < 0)
+            goto error;
+
+        CUDA_EXTERNAL_MEMORY_BUFFER_DESC buf_desc = {
+            .offset = fd_info.offset,
+            .size = fd_info.size,
+            .flags = 0,
+        };
+        ret = CHECK_CU(cu->cuExternalMemoryGetMappedBuffer(&ebuf->buf, ebuf->mem, &buf_desc));
+        if (ret < 0)
+            goto error;
+
+        ra_vk_buf_set_priv(buf, ebuf);
+        MP_DBG(mapper, "Pool %d now has %d buffers\n", n, pool->num_buffers);
```

Comment:
We could probably move this MP_DBG message to ra_buf_pool_get

Reply:
There's actually already a print in ra_buf_pool_grow that I didn't know about. So I'll just remove this.